src/models/entities/Challenge.ts

```diff
@@ -69,9 +71,41 @@ export class Challenge extends BaseEntity implements IChallenge {
   )
   ecoverse?: Ecoverse;
 
+  // The restricted group names at the challenge level
+  @Column('simple-array')
+  restrictedGroupNames?: string[];
+
   constructor(name: string) {
     super();
     this.name = name;
+    this.context = new Context();
+    this.restrictedGroupNames = [RestrictedGroupNames.Members, RestrictedGroupNames.Admins];
+  }
+
+  // Helper method to ensure all members are initialised properly.
+  // Note: has to be a seprate call due to restrictions from ORM.
+  initialiseMembers(): Challenge {
```

Comment:
Move to a service, contains business logic.

Reply:
Split out of business logic should be part of the refactoring.